src/core/Trigger.helper.js

```diff
@@ -1,54 +1,83 @@
 /* eslint-disable no-plusplus */
 /* eslint no-await-in-loop: 0 */
 export default class TriggerHelper {
+
+  static onLoadTriggered = false;
+
+  /** @type {number} - To be used within {@link WaitAndTriggerOptions.interval} */
+  static defaultInterval = 10;
+  /** @type {number} - To be used within {@link WaitAndTriggerOptions.timeout} */
+  static defaultTimeout = 1000;
+
   static async timeout (ms) {
     return new Promise(r => setTimeout(r, ms));
   }
 
   static async action (producer) {
     try {
       await producer.collect();
-    } catch (cause) {
+    } catch (error) {
+      console.log(error);
       // Failed to process event
     }
   }
 
-  static async defaultCondition () {
-    try {
-      const perf = !!(window.performance && window.performance.timing && ((window.performance.timing.domComplete - window.performance.timing.navigationStart) > 0));
-
-      let paintReady = false;
-      const po = new PerformanceObserver((entryList) => {
-        paintReady = entryList.getEntries().length > 0;
+  static async windowOnload () {
+    if (TriggerHelper.onLoadTriggered === true) {
+      return true;
+    } else {
+      return new Promise((resolve, reject) => {
+        window.onload = () => {
+          TriggerHelper.onLoadTriggered = true;
+          resolve(TriggerHelper.onLoadTriggered);
+        };
       });
+    }
+  }
 
-      // Observe entries of type `largest-contentful-paint`, including buffered entries,
-      // i.e. entries that occurred before calling `observe()` below.
-      po.observe({
-        entryTypes: ['paint']
-      });
+  static async defaultCondition () {
+    try {
+      const legacyPerf = !!(window.performance && window.performance.timing
+          && ((window.performance.timing.domComplete - window.performance.timing.navigationStart) > 0));
+      let newPerf = performance.getEntriesByType('navigation')[0];
+      newPerf = ((newPerf.domComplete - newPerf.startTime) > 0);
+      const documentReady = (document.readyState === 'complete' || document.readyState === 'interactive');
+      const onloadTriggered = await TriggerHelper.windowOnload();
```

Comment:
@robertdumitrescu BUG - This will add a new event handler for `onload` every time it is called.

Reply:
OK - Done